tokio/src/net/tcp/listener.rs

```diff
@@ -240,6 +240,32 @@ impl TcpListener {
         Ok(TcpListener { io })
     }
 
+    /// Create a new TcpListener with the provided raw epoll flags.
+    ///
+    /// These flags replace any epoll flags would normally set when registering the fd.
+    ///
+    /// # Note
+    /// This API does not support the use of `EPOLLONESHOT`.
+    /// Users are strongly advised to use `EPOLLET` to prevent the tokio IO driver from receiving
+    /// spurious wakes.
+    ///
+    /// # Stability
+    /// This is an [unstable API][unstable]. The public API of this may break in 1.x releases.
+    /// See [the documentation on unstable features][unstable] for details.
+    ///
+    ///  [unstable]: crate#unstable-features
+    #[track_caller]
+    #[cfg(all(target_os = "linux", tokio_unstable))]
+    #[cfg_attr(docsrs, doc(cfg(all(tokio_unstable, target_os = "linux"))))]
+    pub fn from_std_with_epoll_flags(
```

Comment:
I would rather not have these methods on `TcpListener`. Consider adding them to `TcpSocket` instead as that is the socket builder API.